subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/PerformanceTestBuildExperimentRunner.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.fixture;
+
+import org.gradle.api.Action;
+import org.gradle.internal.concurrent.CompositeStoppable;
+import org.gradle.performance.measure.MeasuredOperation;
+import org.gradle.performance.results.MeasuredOperationList;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class PerformanceTestBuildExperimentRunner extends AbstractBuildExperimentRunner {
```

Comment:
I think it would make a lot of sense to add some documentation here and on the sibling class to explain why we have two implementations, and that this one is going away in favor of the new one.

We could consider adding a prefix here as well like `Legacy` or `GradleInternal` or whatever. Same for `AbstractCrossVersion[Legacy|GradleInternal]PerformanceTest`.

Reply:
I added some documentation and rename the "old" classes to use the prefix `GradleInternal`.

Reply:
Thanks.